packages/pricing/src/services/pricing-module.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  Context,
+  DAL,
+  FindConfig,
+  InternalModuleDeclaration,
+  JoinerServiceConfig,
+  PricingTypes,
+} from "@medusajs/types"
+import { Currency } from "@models"
+import { CurrencyService } from "@services"
+
+import {
+  InjectManager,
+  InjectTransactionManager,
+  MedusaContext,
+} from "@medusajs/utils"
+
+import { shouldForceTransaction } from "@medusajs/utils"
+import { joinerConfig } from "../joiner-config"
+
+type InjectedDependencies = {
+  baseRepository: DAL.RepositoryService
+  currencyService: CurrencyService<any>
+}
+
+export default class PricingModuleService<TCurrency extends Currency = Currency>
+  implements PricingTypes.IPricingModuleService
+{
+  protected baseRepository_: DAL.RepositoryService
+  protected readonly currencyService_: CurrencyService<TCurrency>
+
+  constructor(
+    { baseRepository, currencyService }: InjectedDependencies,
+    protected readonly moduleDeclaration: InternalModuleDeclaration
+  ) {
+    this.baseRepository_ = baseRepository
+    this.currencyService_ = currencyService
+  }
+
+  __joinerConfig(): JoinerServiceConfig {
+    return joinerConfig
+  }
+
+  @InjectManager("baseRepository_")
+  async retrieveCurrency(
+    code: string,
+    config: FindConfig<PricingTypes.CurrencyDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<PricingTypes.CurrencyDTO> {
+    const currency = await this.currencyService_.retrieve(
+      code,
+      config,
+      sharedContext
+    )
+
+    return JSON.parse(JSON.stringify(currency))
```

Comment:
It would be nice to replace all the `JSON.parse(JSON.stringify())` with `this.baseRepository_.serialize()`
That will better handle data types like Date...